src/formatDate.js

```diff
@@ -1,14 +1,50 @@
 'use strict';
 
-/**
- * @param {string} date
- * @param {string[]} fromFormat
- * @param {string[]} toFormat
- *
- * @returns {string}
- */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const dateArr = date.split(oldSeparator);
+
+  const dayIndex = fromFormat.indexOf('DD');
+  const monthIndex = fromFormat.indexOf('MM');
+  const yearIndex = fromFormat.findIndex(
+    (format) => format === 'YY' || format === 'YYYY',
+  );
+
+  const day = dateArr[dayIndex];
+  const month = dateArr[monthIndex];
+  let year = dateArr[yearIndex];
+
+  const currentYear = new Date().getFullYear() % 100;
+
+  if (fromFormat[yearIndex] === 'YY') {
+    if (parseInt(year) <= currentYear) {
+      year = '20' + year.padStart(2, '0');
+    } else {
+      year = '19' + year.padStart(2, '0');
+    }
+  }
```

Comment:
Move all formats (YYYY,YY) to constant. 
Also move all number values (20, 19) to constant